test/pool_test.cc

```diff
@@ -0,0 +1,119 @@
+#include <chrono>
+
+#include "common/milestone_tracker_impl.h"
+#include "common/pool_impl.h"
+#include "common/poolable_impl.h"
+
+#include "test/mocks.h"
+#include "test/test_common/simulated_time_system.h"
+
+#include "gtest/gtest.h"
+
+using namespace testing;
+
+namespace Nighthawk {
+
+class MockPoolablePoolImpl : public PoolImpl<MockPoolable> {
```

Comment:
As above, probably doesn't need to be a class.

Reply:
fixed